res/controllers/Pioneer-XDJ-RX-scripts.js

```diff
@@ -0,0 +1,255 @@
+    engine.softTakeover("[Channel2]", "rate", true);
+    engine.softTakeover("[Channel1]", "cue_default", true);
+    engine.softTakeover("[Channel2]", "cue_default", true);
+    engine.softTakeover("[Channel1]", "jog", true);
+    engine.softTakeover("[Channel2]", "jog", true);
+    engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[Master]", "volume", true);
+    engine.softTakeover("[Channel1]", "play", true);
+    engine.softTakeover("[Channel2]", "play", true);
+    engine.softTakeover("[Channel1]", "SelectTrackKnob", true);
+    engine.softTakeover("[Channel2]", "SelectTrackKnob", true);
+    engine.softTakeover("[Channel1]", "keylock", true);
+    engine.softTakeover("[Channel2]", "keylock", true);
+    engine.softTakeover("[Channel1]", "beatsync", true);
+    engine.softTakeover("[Channel2]", "beatsync", true);
+    engine.softTakeover("[Playlist]", "ToggleSelectedSidebarItem", true);
+}
+PioneerXDJRX.shutdown = function(id) {}
+
+PioneerXDJRX.reverse_toggle = function(channel, control, value, status, group) {
+    if (value == 0)
+        return;
+    if (PioneerXDJRX.reverse_play)
+        PioneerXDJRX.reverse_play = false;
+    else
+        PioneerXDJRX.reverse_play = true;
+    engine.setValue(group, 'reverse', PioneerXDJRX.reverse_play);
+}
+
+PioneerXDJRX.tempo_btn = function(channel, control, value, status, group) {
+    if (value == 0x00) {
+        return;
+    }
+
+    oldValue = engine.getValue(group, "rateRange");
+    newValue = 0.06;
+
+    if (oldValue > 0.11) {
+        newValue = 0.03;
+    } else if (oldValue > 0.05) {
+        newValue = 0.12;
+    }
+
+    engine.setValue(group, "rateRange", newValue);
+};
+
+PioneerXDJRX.jog_wheel = function(channel, control, value, status, group) {
+    value = (value - 0x40) / PioneerXDJRX.jog_sensitivity;
+    engine.setValue(group, "jog", value);
+
+};
+//Inactive
+PioneerXDJRX.jog_scratch = function(channel, control, value, status, group) {
+    value = (value - 0x40) / PioneerXDJRX.scratch_sensitivity;
+    engine.setValue(group, 'jog', value);
+
+};
+
+PioneerXDJRX.ClearCue1 = function(channel, control, value, status, group) {
+    engine.setValue("[Channel1]", "cue_point", 0);
+}
+
+PioneerXDJRX.ClearCue2 = function(channel, control, value, status, group) {
+    engine.setValue("[Channel2]", "cue_point", 0);
+}
+
+PioneerXDJRX.select_track_knob = function(channel, control, value, status, group) {
+    if (value >= 0x01 && value <= 0x1e) {
+        value = value;
+    } else if (value >= 0x62 && value <= 0x7F) {
+        value = 0 - (0x7F - value + 1);
+    } else {
+        return;
+    }
+    engine.setValue(group, 'SelectTrackKnob', value);
+};
+
+var headPhone1On = false;
+var headPhone1lastclicked;
+PioneerXDJRX.toogleHeadPhone1 = function(channel, control, value, status) {
+    var thisClick = new Date().getTime();
+    if ((headPhone1lastclicked + 250) > thisClick) {
+        return;
+    }
```

Comment:
i guess to prevent a situation when one presses a mouse button but two clicks are generated.

Reply:
Can you remove the clause and test whether that causes issues?

Reply:
yes, it causes issues.

when the code looks like:

```
var headPhone1On = false;
var headPhone1lastclicked;
PioneerXDJRX.toogleHeadPhone1 = function(channel, control, value, status) {
//    var thisClick = new Date().getTime();
//    if ((headPhone1lastclicked + 250) > thisClick) {
//        return;
//    }
//    headPhone1lastclicked = thisClick;
    if (!headPhone1On) {
        midi.sendShortMsg(0x96, 0x54, 127); //headphone button 1 lights on
        engine.setValue("[Channel1]", "pfl", 1);
        headPhone1On = true;
    } else {
        midi.sendShortMsg(0x96, 0x54, 0); //headphone button 1 lights off
        engine.setValue("[Channel1]", "pfl", 0);
        headPhone1On = false;
    }
};
```

then when i press the headphone button, only by keeping it pressed i can hear the sound in the headphones, otherwise headphone sound is not enabled.